core/src/main/scala/tech/beshu/ror/accesscontrol/factory/JsonConfigStaticVariableResolver.scala

```diff
@@ -54,17 +54,31 @@ object JsonConfigStaticVariableResolver {
       .mapObject(_.mapValues(mapJson(_, errors)))
       .withString { str =>
         val resolved = NonEmptyString.unapply(str) match {
-          case Some(nes) => tryToResolveAllStaticSingleVars(nes, errors)
+          case Some(nes) =>
+            tryToResolveAllStaticSingleVars(nes, errors)
           case None => str
         }
-        resolvedStringToJson(resolved)
+        if (resolved =!= str)
+          resolvedStringToJson(resolved, json)
+        else json
       }
   }
 
-  private def resolvedStringToJson(resolvedStr: String) = {
+  private def preserveNumbersAsStrings(original: Json, newValue: Json) = {
+    {
+      for {
+        _ <- original.asString
```

Comment:
I don't know if I understand what is going on here. We try to get a string from JSON and ignore the result and then try to get a number from the new value and encode it as a string or if all of these fail we return the newValue. It's rather hacky, so we should do one of the following:
1. write more declarative or at least descriptive function
2. explain in comment the intention of the hack

Reply:
Nevermind, I played type tetris a bit too long when I did this and I was brain dead. I simplified the algorithm now and it should be evident what's happening.